molgenis/molgenis-emx2
# feat: show tree input in forms and filters for 'ontology tree' like data.

Motivation: we want to show CollectionEvents.AreasOfInformation as tree input in forms, so cohort data managers can more easily select the appropriate element as compared to current list,  and in data explorer, so researchers can more easily set filters.

Therefore added 'ontology tree'. Also as side effect, merged 'columnFormat' and 'columnType' into one parameter, because having two will lead to errors.

How to test
- in styleguide view, take a look at OntologyTerm component. Notice it is quite responsive.
- in schema designer you can now enter a columnFormat to Collectionevent.areasOfInformation column
- in molgenis.csv of cohort catalogue set columnFormat=OntologyTerm for column=AreasOfInformation to do this during upload
- in forms you can now select areasOfInformation in a tree
- in data explorer you can select filter for areasOfInformation in a tree

Known limitations: this component should be limited to ontologies that are not too large. E.g. for ICD10 we should think to make it more 'lazy'.

![image](https://user-images.githubusercontent.com/120060/130358142-5ec5f0fa-399d-4ce1-8ebd-137e5a3aa1ab.png)
